src/htmldiff.ts

```diff
@@ -184,19 +216,25 @@ export function htmlToTokens(html: string): Token[] {
   let mode = 'char';
```

Comment:
Lets make this a union type of all the possible `mode`s so that we can ensure we are addressing all of them in this switch by adding an `exhaustive` function at the end of this function.